Symfony/CS/Fixer/Symfony/PhpUnitDedicateAssertFixer.php

```diff
@@ -0,0 +1,216 @@
+    {
+        return 'PHPUnit assertions like "assertInternalType", "assertFileExists", should be used over "assertTrue". Warning! This could change code behavior.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run after the PhpUnitConstructFixer.
+        return -15;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $assertCallIndex Token index of assert method call.
+     *
+     * @return int|int[] indexes of assert call, test call and positive flag, or last index checked.
+     */
+    private function getAssertCandidate(Tokens $tokens, $assertCallIndex)
+    {
+        $content = strtolower($tokens[$assertCallIndex]->getContent());
+        if ('asserttrue' === $content) {
+            $isPositive = 1;
+        } elseif ('assertfalse' === $content) {
+            $isPositive = 0;
+        } else {
+            return $assertCallIndex;
+        }
+
+        // test candidate for simple calls like: ([\]+'some fixable call'(...))
+        $assertCallOpen = $tokens->getNextMeaningfulToken($assertCallIndex);
+        if (!$tokens[$assertCallOpen]->equals('(')) {
+            return $assertCallIndex;
+        }
+
+        $testNsToken = false;
+        $test = $tokens->getNextMeaningfulToken($assertCallOpen);
+
+        if (!$tokens[$test]->isGivenKind(T_EMPTY) && !$tokens[$test]->isGivenKind(T_STRING)) {
+            if (!$tokens[$test]->isGivenKind(T_NS_SEPARATOR) || '\\' !== $tokens[$test]->getContent()) {
+                return $test;
+            }
+
+            $testNsToken = $test;
+            $test = $tokens->getNextMeaningfulToken($test);
+            if (!$tokens[$test]->isGivenKind(T_STRING)) {
+                return $test;
+            }
+        }
+
+        $testOpen = $tokens->getNextMeaningfulToken($test);
+        if (!$tokens[$testOpen]->equals('(')) {
+            return $testOpen;
+        }
+
+        $testClose = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $testOpen);
+
+        $assertCallClose = $tokens->getNextMeaningfulToken($testClose);
+        if (!$tokens[$assertCallClose]->equals(')') && !$tokens[$assertCallClose]->equals(',')) {
+            return $assertCallClose;
+        }
+
+        return array(
+            $isPositive,
+            $assertCallIndex,
+            $assertCallOpen,
+            $testNsToken,
+            $test,
+            $testOpen,
+            $testClose,
+            $assertCallClose,
+        );
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param array  $assertIndexes
+     *
+     * @return int index up till processed, number of tokens added
+     */
+    private function fixAssert(Tokens $tokens, array $assertIndexes)
+    {
+        list(
+            $isPositive,
+            $assertCallIndex,
+            $assertCallOpen,
+            $testNsToken,
+            $test,
+            $testOpen,
+            $testClose,
+            $assertCallClose) = $assertIndexes;
```

Comment:
thanks, also for all other great pointers! I've made changes following your suggestions :)